tests/src/Mod/Part/App/TopoShapeExpansion.cpp

```diff
@@ -385,6 +385,120 @@ TEST_F(TopoShapeExpansionTest, MapperHistoryGenerated)
     EXPECT_EQ(fuse2MprHst.generated(edge3).size(), 1);  // fuse2 has a new vertex generated by edge3
 }
 
+TEST_F(TopoShapeExpansionTest, resetElementMapTest)
+{
+    // Arrange
+    // Creating various TopoShapes to check different conditions
+
+    // A TopoShape without a map
+    auto shapeWithoutMap {
+        TopoShape(BRepBuilderAPI_MakeEdge(gp_Pnt(-1.0, 0.0, 0.0), gp_Pnt(1.0, 0.0, 0.0)).Edge(),
+                  1)};
+
+    // A TopoShape without a map that will be replaced by another map
+    auto shapeWithoutMapAfterReset {TopoShape(shapeWithoutMap)};
+
+    // A TopoShape with a map
+    auto shapeWithMap {
+        TopoShape(BRepBuilderAPI_MakeEdge(gp_Pnt(0.0, -1.0, 0.0), gp_Pnt(0.0, 1.0, 0.0)).Edge(),
+                  3)};
+    shapeWithMap.makeShapeWithElementMap(shapeWithMap.getShape(),
+                                         TopoShape::Mapper(),
+                                         {shapeWithMap});
+
+    // A TopoShape with a map that will be replaced by another map
+    auto shapeWithMapAfterReset {TopoShape(shapeWithMap)};
+    shapeWithMapAfterReset.makeShapeWithElementMap(shapeWithMapAfterReset.getShape(),
+                                                   TopoShape::Mapper(),
+                                                   {shapeWithMapAfterReset});
+
+    // A TopoShape with a map that will be replaced by an empty map
+    auto shapeWithMapAfterEmptyReset {TopoShape(shapeWithMap)};
+    shapeWithMapAfterEmptyReset.makeShapeWithElementMap(shapeWithMapAfterEmptyReset.getShape(),
+                                                        TopoShape::Mapper(),
+                                                        {shapeWithMapAfterEmptyReset});
+
+    // A new map
+    auto newElementMapPtr {std::make_shared<Data::ElementMap>()};
+    newElementMapPtr->setElementName(IndexedName("Edge", 2),
+                                     MappedName("Edge2;:H,E"),
+                                     3,
+                                     nullptr,
+                                     true);
+
+    // Act
+    shapeWithoutMapAfterReset.resetElementMap(newElementMapPtr);
+    shapeWithMapAfterReset.resetElementMap(newElementMapPtr);
+    shapeWithMapAfterEmptyReset.resetElementMap(nullptr);
+
+    // Assert
+    // Check that the original maps haven't been modified
+    EXPECT_EQ(shapeWithoutMap.getElementMapSize(false), 0);
+    EXPECT_EQ(shapeWithMap.getElementMapSize(false), 3);
+
+    // Check that the two shapes have the same map
+    EXPECT_EQ(shapeWithoutMapAfterReset.getElementMap(), shapeWithoutMapAfterReset.getElementMap());
```

Comment:
Should both sides have the same clause here?

Reply:
Looks I missed something along the way, originally I was checking `shapeWithoutMapAfterReset` and `shapeWithMapAfterReset` (I guess, trying to remember).

They should have the same map as both reset with `newElementMapPtr`.

Fixing it in the next commit